### PR TITLE
add `h_glm_negbin()`

### LINKED ISSUE #856: Add `h_glm_negbin()`
Start of negative binomial functionality.

To do:
- [x] See [`h_glm_poisson()`](https://github.com/insightsengineering/tern/blob/main/R/summarize_glm_count.R#L24) to see how it works
- [x] Then implement `h_glm_negbin()`in https://github.com/insightsengineering/tern/blob/main/R/summarize_glm_count.R
	- [x] Use `MASS::glm.nb()`
- [x] Add tests in https://github.com/insightsengineering/tern/blob/main/tests/testthat/test-summarize_glm_count.R

### PR DESCRIPTION
**Pull request by duanx9**
Close #856 

- Implement `h_glm_negbin()`.
- Add unit tests.
- Update `NEWS.md`.

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -38,6 +38,7 @@ Imports:
     labeling,
     lifecycle (>= 0.2.0),
     magrittr (>= 1.5),
+    MASS,
```
Comment (shajoezhu) [inline DESCRIPTION:41]: perhaps provide a minimal version (use the latest tag from cran would be fine)

File: R/summarize_glm_count.R [hunk 2]
```diff
@@ -156,13 +200,10 @@ h_glm_count <- function(.var,
                         variables,
                         distribution,
                         weights) {
-  if (distribution == "negbin") {
-    stop("negative binomial distribution is not currently available.")
-  }
```
Comment (shajoezhu) [inline R/summarize_glm_count.R:161]: adding assertion to check if `distribution %in% c("poisson", "quasiposson", "negbin")`

File: tests/testthat/test-summarize_glm_count.R [hunk 3]
```diff
@@ -107,6 +107,60 @@ testthat::test_that("h_glm_quasipoisson fails wrong inputs", {
...
+testthat::test_that("h_glm_negbin glm-fit works with healthy input", {
+  anl <- tern_ex_adtte %>%
+    filter(PARAMCD == "TNE")
+  anl$AVAL_f <- as.factor(anl$AVAL)
+
+  result <- h_glm_negbin(
+    .var = "AVAL",
+    .df_row = anl,
+    variables = list(arm = "ARM", offset = "lgTMATRSK", covariates = c("REGION1"))
+  )
```
Comment (shajoezhu) [inline tests/testthat/test-summarize_glm_count.R:119]: I think it would also be nice to include some example code

e.g.
```
  anl <- tern_ex_adtte %>%
    filter(PARAMCD == "TNE")
  anl$AVAL_f <- as.factor(anl$AVAL)

# Poisson
h_glm_poisson(
    .var = "AVAL",
    .df_row = anl,
    variables = list(arm = "ARM", offset = "lgTMATRSK", covariates = c("REGION1"))
  )


# Negative binomial
h_glm_negbin(
    .var = "AVAL",
    .df_row = anl,
    variables = list(arm = "ARM", offset = "lgTMATRSK", covariates = c("REGION1"))
  )



```

**Review by shajoezhu — CHANGES_REQUESTED** [review]
Thanks @duanx9 ! I have made some small comments. Cheers

**Review by shajoezhu — APPROVED** [review]
lgtm! Thanks @duanx9

**Comment by shajoezhu** [comment]
hang on! @duanx9 , let me take another look

**Review by shajoezhu — CHANGES_REQUESTED** [review]
hi @duanx9 
I was wondering if you can add more test cases (snapshots), for the negative binomial model, take examples from 

1. testthat::test_that("summarize_glm_count works with healthy inputs", {
2. testthat::test_that("s_glm_count works with no reference group selected.", {
3. testthat::test_that("s_glm_count works with healthy input", {

rename previous test with "poisson distribution", and add new ones for "negative binomial"

**Review by shajoezhu — APPROVED** [review]
Brilliant! Many thanks @duanx9
